swagger-generator/src/main/java/com/mrv/yangtools/codegen/impl/AbstractDataObjectBuilder.java

```diff
@@ -250,6 +252,16 @@ protected Pair prop(DataSchemaNode node) {
         } else if (node instanceof LeafSchemaNode) {
             LeafSchemaNode lN = (LeafSchemaNode) node;
             prop = getPropertyByType(lN);
+            if (prop != null) {
```

Comment:
why did you decided to set required only for simple fields? Swagger allows you to define it at any property as well as YANG defines it at `DataSchemaNode` so it can be applied for all types of `node`s